olp-cpp-sdk-dataservice-read/src/Common.h

```diff
@@ -58,6 +60,35 @@ inline client::CancellationToken ScheduleFetch(TaskScheduler&& schedule_task,
                        std::forward<Callback>(callback));
 }
 
+/*
+ * @brief Common function used to wrap a lambda function and a callback that
+ * consumes the function result with a TaskContext class and schedule this to a
+ * task scheduler.
+ * @param task_scheduler Task scheduler instance.
+ * @param pending_requests PendingRequests instance that tracks current
+ * requests.
+ * @param task Function that will be executed.
+ * @param callback Function that will consume task output.
+ * @param args Additional agrs to pass to TaskContext.
+ * @return CancellationToken used to cancel the operation.
+ */
+template <typename Function, typename Callback, typename... Args>
+inline client::CancellationToken AddTask(
+    std::shared_ptr<thread::TaskScheduler> task_scheduler,
+    std::shared_ptr<client::PendingRequests> pending_requests, Function task,
```

Comment:
As this is a sync call, I would pass them as reference, were PendingRequest si mandatory:

```cpp
inline client::CancellationToken AddTask(
    std::shared_ptr<thread::TaskScheduler>& task_scheduler,
    client::PendingRequests& pending_requests, Function task,
```